rclpy/src/rclpy/_rclpy.c

```diff
@@ -3193,28 +3540,48 @@ rclpy_get_service_names_and_types(PyObject * Py_UNUSED(self), PyObject * args)
   }
 
   PyObject * pyservice_names_and_types = PyList_New(service_names_and_types.names.size);
+  if (!pyservice_names_and_types) {
+    goto cleanup;
+  }
   size_t i;
   for (i = 0; i < service_names_and_types.names.size; ++i) {
     PyObject * pytuple = PyTuple_New(2);
-    PyTuple_SetItem(
-      pytuple, 0,
-      PyUnicode_FromString(service_names_and_types.names.data[i]));
-    PyObject * types_list = PyList_New(service_names_and_types.types[i].size);
+    if (!pytuple) {
+      goto cleanup;
+    }
+    PyObject * pyservice_name = PyUnicode_FromString(service_names_and_types.names.data[i]);
+    if (!pyservice_name) {
+      Py_DECREF(pytuple);
+      goto cleanup;
+    }
+    PyTuple_SetItem(pytuple, 0, pyservice_name);
+    PyObject * pytypes_list = PyList_New(service_names_and_types.types[i].size);
+    if (!pytypes_list) {
+      Py_DECREF(pytuple);
+      Py_DECREF(pyservice_name);
```

Comment:
@jacobperron I _think_ `PyTuple_SetItem` on [line 3557](https://github.com/ros2/rclpy/pull/278/files#diff-80dda03110b5be606b823d6b0558b5c2R3557) steals `pyservice_name` reference.

Reply:
Good catch. Fixed in 914ff23